extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/CredentialsVerifierExtension.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright (c) 2021 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.verifier;
+
+import okhttp3.OkHttpClient;
+import org.eclipse.dataspaceconnector.iam.did.spi.credentials.CredentialsVerifier;
+import org.eclipse.dataspaceconnector.iam.did.spi.resolution.DidPublicKeyResolver;
+import org.eclipse.dataspaceconnector.identityhub.client.IdentityHubClientImpl;
+import org.eclipse.dataspaceconnector.identityhub.credentials.VerifiableCredentialsJwtUtils;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.system.Provider;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+
+import static java.lang.String.format;
+
+/**
+ * Extension to provide verification of IdentityHub Verifiable Credentials.
+ */
+public class CredentialsVerifierExtension implements ServiceExtension {
+
+    @EdcSetting
+    private static final String HUB_URL_SETTING = "edc.identity.hub.url";
+
+    @Inject
+    private OkHttpClient httpClient;
+
+    @Inject
+    private DidPublicKeyResolver didPublicKeyResolver;
+
+    @Inject
+    private TypeManager typeManager;
+
+    @Inject
+    private Monitor monitor;
+
+    @Inject
+    private JwtCredentialsVerifier jwtCredentialsVerifier;
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        monitor.info("Initialized Identity Hub DID extension");
+    }
+
+    @Provider
+    public CredentialsVerifier createCredentialsVerifier(ServiceExtensionContext context) {
+        var hubUrl = context.getSetting(HUB_URL_SETTING, null);
+        if (hubUrl == null) {
+            throw new EdcException(format("Mandatory setting '(%s)' missing", HUB_URL_SETTING));
```

Comment:
should be moved to initialise method (so this exception is raised at the expected point in the lifecycle)

Reply:
@paullatzelsperger What is your take on this? If @algattik is correct we should probably document this [here](https://github.com/agera-edc/DataSpaceConnector/blob/main/docs/developer/dependency_resolution.md#option-1-use-provider-methods-recommended)?

Reply:
For context leaving it like it is throws an Exception during Registration phase. For debuggability it makes not much difference, I guess it would be more a question of correctness?
```
SEVERE 2022-07-25T09:23:18.38204 Error booting runtime
org.eclipse.dataspaceconnector.spi.system.injection.EdcInjectionException: java.lang.reflect.InvocationTargetException
	at org.eclipse.dataspaceconnector.spi.system.injection.ProviderMethod.invoke(ProviderMethod.java:66)
	at org.eclipse.dataspaceconnector.boot.system.injection.lifecycle.RegistrationPhase.invokeAndRegister(RegistrationPhase.java:51)
...
Caused by: org.eclipse.dataspaceconnector.spi.EdcException: Mandatory setting '(edc.identity.hub.url)' missing
```